File: firstOccurrence/firstOccurrence.py
def first_occurrence(arr: list[int], target: int) -> int | None:
    if not arr:
        return None

    left, right = 0, len(arr) - 1

    while left < right:
        middle = (left + right) // 2

        if arr[middle] < target:
            left = middle + 1
        else:
            right = middle

    return left if arr[left] == target else None

File: firstOccurrence/firstOccurrence.ts
function firstOccurrence(arr: number[], numb: number): number | null {
  if (arr.length === 0) {
    return null;
  }

  let left = 0;
  let right = arr.length - 1;

  while (left < right) {
    const middle = left + Math.floor((right - left) / 2);

    if (arr[middle] < numb) {
      left = middle + 1;
    } else {
      right = middle;
    }
  }

  return arr[left] === numb ? left : null;
}